src/common/darwin_errno.cc

```diff
@@ -0,0 +1,224 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <errno.h>
+#include "include/types.h"
+#include "include/compat.h"
+
+
+// converts from linux errno values to host values
+__s32 ceph_to_host_errno(__s32 r)
+{
+  if (r < -34) {
+    switch (r) {
+      case -35:
+        return -EDEADLK;
+      case -36:
+        return -ENAMETOOLONG;
+      case -37:
+        return -ENOLCK;
+      case -38:
+        return -ENOSYS;
+      case -39:
+        return -ENOTEMPTY;
+      case -40:
+        return -ELOOP;
+      case -42:
+        return -ENOMSG;
+      case -43:
+        return -EIDRM;
+      case -44:
+        return -EPERM; //TODO ECHRNG
+      case -45:
+        return -EPERM; //TODO EL2NSYNC
+      case -46:
+        return -EPERM; //TODO EL3HLT
+      case -47:
+        return -EPERM; //TODO EL3RST
+      case -48:
+        return -EPERM; //TODO ELNRNG
+      case -49:
+        return -EPERM; //TODO EUNATCH
+      case -51:
+        return -EPERM; //TODO EL2HLT;
+      case -52:
+        return -EPERM; //TODO EBADE
+      case -53:
+        return -EPERM; //TODO EBADR
+      case -54:
+        return -EPERM; //TODO EXFULL
+      case -55:
+        return -EPERM; //TODO ENOANO
+      case -56:
+        return -EPERM; //TODO EBADRQC
+      case -57:
+        return -EPERM; //TODO EBADSLT
+      case -59:
+        return -EPERM; //TODO EBFONT
+      case -60:
+        return -ENOSTR;
+      case -61:
+        return -ENOATTR;
```

Comment:
The first indication that something is amiss  is running the unittest_*attr* stuff.
There is a loop in the attr code that tests on ENODATA but the call returns ENOATTR (on FreeBSD)
And the only exit out of that loop is the get ENODATA, so the program is stuck in this loop.